connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -255,7 +255,23 @@ public ConfigInfos validateConnectorConfig(Map<String, String> connectorProps) {
 
             // do custom connector-specific validation
             Config config = connector.validate(connectorProps);
+            if (null == config) {
+                throw new BadRequestException(
+                    String.format(
+                        "%s.validate() must return a valid Config.",
+                        connector.getClass().getName()
+                    )
+                );
+            }
             ConfigDef configDef = connector.config();
+            if (null == configDef) {
+                throw new BadRequestException(
+                    String.format(
+                        "%s.config() must return a valid ConfigDef.",
```

Comment:
Similar question here -- what about:

    %s.config() must return a ConfigDef that is not null.

like on [this line](https://github.com/apache/kafka/pull/3762/files#diff-99fd9ec8598ede0037750255a669db4bR233)?